main.js

```diff
@@ -101,22 +110,39 @@ function getRandomHex() {
     return (Math.floor(Math.random() * 16777216).toString(16).padStart(6, 0)).toUpperCase();
 }
 
-function savePalette() {
+async function savePalette() {
+    modalClassToggler([[saveModal, 'hidden'], [main, 'block'], [savedContainer, 'block']]);
     if (!savedPalettes.length) {
         paragraph.classList.add('hidden');
     } 
     if (isPaletteUnique(savedPalettes, currentPalette)) {
+        await getPromiseFromEvent2(saveModalButton, 'click');
         savedPalettes.push(currentPalette);
         addPaletteToSavedPalettes(currentPalette);
     }
+    modalClassToggler([[saveModal, 'hidden'], [main, 'block'], [savedContainer, 'block']]);
     getNewHexes(mainColorBoxes);
 }
 
-function isPaletteUnique(palettesList, singlePalette) {
-    for (i = 0; i < palettesList.length; i++) {
+function getPromiseFromEvent2(element, listenerName) {
+    return new Promise(function (resolve) {
+        function listener(event) {
+            if (event.target.nodeName === 'BUTTON') {
+                currentPalette.name = saveModalInput.value;
+                saveModalInput.value = ''
+            }
+            element.removeEventListener(listenerName, listener);
+            resolve(event);
+        };
+        element.addEventListener(listenerName, listener);
+    });
+}
+
+function isPaletteUnique(savedPalettes, currentPalette) {
+    for (i = 0; i < savedPalettes.length; i++) {
         var matches = true;
-        for (j = 0; j < palettesList[i].length; j++) {
-            if (palettesList[i][j] !== singlePalette[j]) {
+        for (j = 0; j < savedPalettes[i].hexes.length; j++) {
+            if (savedPalettes[i].hexes[j] !== currentPalette.hexes[j]) {
```

Comment:
Should we (do we even have time) to also check to make sure the name is unique. ie: can have only one saved name?

Reply:
I'm just gonna piggy-back on the uniqueness function @SethBurns already made for us! ❤️